### DISCUSSION TITLE
prefix and suffix question

### DISCUSSION THREAD

**Discussion by aniel300** [euzu/tuliprox#109] [Q&A]
can i get a sample of how this goes? am taking a look at this

```
prefix and suffix are appended after all processing is done, but before sort. They have 2 fields:

field can be name , group, title
value a static text
```

but idk how it goes within the compose file

**Comment by euzu** [comment]
This has nothing to do with the compose file.
It is a configuration. mapping.yml is the configuration for defiining the transformations.

When you run m3u-filter it load the configuration files and does what is configured.

**Reply by aniel300** [reply]
sorry i didn't mean to say compose file, i meant to say "source.yml"

**Comment by euzu** [comment]
here is a wiki link [mapping](https://github.com/euzu/m3u-filter/wiki/mapping)

**Reply by aniel300** [reply]
as per this. doesnt the prefix and suffix part goes within imput or targets? am a little confuse
![image](https://github.com/user-attachments/assets/624f23c6-d3ba-434a-84ea-20119779cb2d)

**Comment by euzu** [comment]
Yes this is something different.
When you have more than one input, you can mark the entries with a prefix or suffix to see from where they are comming.

This is applied before the other transformations during input phase.

I dont have any usable use case for this. It was requested by someone.

**Reply by aniel300** [reply]
yep this is something i want. i just need to know the format as how it goes. i have an idea but i want to make sure

**Comment by euzu** [comment]
iam currently not at my computer.
The wiki/readme is confusing.
I will check this tomorrow.

**Reply by aniel300** [reply]
thank u bud. will keep reading the wiki and readme. hopefully i dont get stuck again. this project seems very deep compared to current proxy am using now

**Reply by aniel300** [reply]
i found a sample to the prefix/suffix question btw. https://github.com/euzu/m3u-filter/issues/71

**Reply by aniel300** [reply]
what is the different between name and title for the suffix stuff and can i use more than one field?

**Reply by euzu** [reply]
That is really a think i dont know why the creators of this formats did. There is a name field and a title field. This is not coming from  m3u-filter. Sometimes they are different, most time the same.
As i can see in the source you can only define one prefix and one suffix for an input.
If you need more fields than one, feel free to make a feature request.

**Reply by aniel300** [reply]
i did not understand your answer, sorry.

**Reply by euzu** [reply]
You asked about name and title field:
Answer: This is the playlist drfinition, not comming from me. Both fields can have different values  but mostly they have the same value. It depends on your provider.

You asked can you have multiple prefix or suffix definitions for inputs?
Answer no, only one for suffix and one for prefix. If you need more than one field than make a feature request.

**Reply by aniel300** [reply]
what am asking is if i can have 1 suffix applied to groups but also streams names. and i also wanted to know what the difference between name and title was. group i do understand.

**Reply by euzu** [reply]
If you use mapping, you can append as many suffixes or prefixes as you want, no limitations.

If you use input config prefix/suffix you can onlu use one field for each.

And name and title:
```
#EXTM3U
#EXTINF:-1 tvg-id="asian.tv" tvg-name="Asian TV",Asian TV (720p)
http://myprovider.tv/asian/ytlive/1234.ts
```
tvg-name is name field
everything after comma is title in this case Asian TV (720p)

**Reply by aniel300** [reply]
ok thanks. let me see what i can do here

**Reply by euzu** [reply]
i forgot one thing, this is only for m3u playlists. XC  has only name or title. Both are the same for XC.

**Reply by aniel300** [reply]
ahh ok perfect thanks

**Comment by euzu** [comment]
my i ask what you are using ?

**Reply by aniel300** [reply]
this but am having some major issues with it at the moment. https://github.com/sonroyaalmerol/m3u-stream-merger-proxy

**Reply by euzu** [reply]
this is different than m3u-filter. The focus of m3u-filter never was the proxy functionality. The main purpose was to extract the playlist you want from different sources.  
And one key feature is no dependencies, single binary with config.
You dont need a database or a runtime environment like python or node.

**Reply by aniel300** [reply]
so i cant use this project as a proxy? i though i could?

**Comment by euzu** [comment]
m3u filter is a proxy. i meant its original.idea was not based on beiing a proxy.
M3u-filter can act as redirect or reverse proxy.
You can configure users for each target and set the proxy type infividually.

**Reply by aniel300** [reply]
that works for me. i just need to figure out things little by little as there is a lot to take in

**Comment by aniel300** [comment]
closing as i think things have explained by @euzu, thank you very much for your help
